easybuild/easyblocks/m/mpich.py

```diff
@@ -130,26 +123,32 @@ def configure_step(self):
 
     # make and make install are default
 
-    def sanity_check_step(self, custom_paths=None):
+    def sanity_check_step(self, custom_paths=None, use_new_libnames=None):
         """
         Custom sanity check for MPICH
         """
         shlib_ext = get_shared_lib_ext()
         if custom_paths is None:
             custom_paths = {}
-                
+
+        if use_new_libnames is None:
+            # cfr. http://git.mpich.org/mpich.git/blob_plain/v3.1.1:/CHANGES
+            # MPICH changed its library names sinceversion 3.1.1
+            use_new_libnames = LooseVersion(self.version) >= LooseVersion('3.1.1')
+
         # Starting MPICH 3.1.1, libraries have been renamed
         # cf http://git.mpich.org/mpich.git/blob_plain/v3.1.1:/CHANGES
-        if self.use_new_libnames():
+        if use_new_libnames:
             libnames = ['mpi', 'mpicxx', 'mpifort']
         else:
             libnames = ['fmpich', 'mpichcxx', 'mpichf90', 'mpich', 'mpl', 'opa']
 
-        dirs = ['include', 'bin', 'lib']
         bins = ['bin/%s' % x for x in ['mpicc', 'mpicxx', 'mpif77', 'mpif90', 'mpiexec.hydra', 'mpirun', 'mpiexec']]
         headers = ['include/%s' % x for x in ['mpi.h', 'mpicxx.h', 'mpif.h']]
-        libs = ['lib/lib%s.%s' % (l,e) for l in libnames for e in ('a', shlib_ext)]
-            
-        custom_paths.setdefault('dirs',  []).extend(dirs)
-        custom_paths.setdefault('files', []).extend(headers + bins + libs)
+        libs = ['lib/lib%s.%s' % (l, e) for l in libnames for e in ('a', shlib_ext)]
+
+        # only set files/dirs keys if they weren't defined yet in custom_paths that was provided
+        custom_paths.setdefault('files', bins + headers + libs)
+        custom_paths.setdefault('dirs', ['bin', 'include', 'lib'])
```

Comment:
About this "only set files/dirs keys if they weren't defined yet in custom_paths that was provided"

Shouldn't we then set the `custom_path` only if its value is `None` (cf line 139/132)?